packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -300,13 +300,11 @@ export const BlockBindingsPanel = ( { name: blockName, metadata } ) => {
 						/>
 					) }
 				</ItemGroup>
-				<ItemGroup>
-					<Text variant="muted">
-						{ __(
-							'Attributes connected to custom fields or other dynamic data.'
-						) }
-					</Text>
-				</ItemGroup>
+				<Text as="p" variant="muted">
+					{ __(
+						'Attributes connected to custom fields or other dynamic data.'
+					) }
+				</Text>
```

Comment:
The fix outlined by @SantosGuillamot in https://github.com/WordPress/gutenberg/pull/67427#issuecomment-2510808820 would look like:
```suggestion
				{ bindings.length && <Text as="p" variant="muted">
					{ __(
						'Attributes connected to custom fields or other dynamic data.'
					) }
				</Text> }
```

Prettier might have some opinions about formatting.

Reply:
Interesting. The ToolsPanle mechanism to toggle the visibility of this item assumes a `div` element is in use:

https://github.com/WordPress/gutenberg/blob/b867a5fe7c1133e906682eff84b53b6621dea559/packages/components/src/tools-panel/styles.ts#L60-L64

I'm not sure I like it and it seems a bit fragile to me.

Reply:
> The fix outlined by @SantosGuillamot in [#67427 (comment)](https://github.com/WordPress/gutenberg/pull/67427#issuecomment-2510808820) would look like:
> 
> Prettier might have some opinions about formatting.

I tested it a bit and it could use `Object.keys( filteredBindings ).length`. However, that would make the help text appear after:
1. An attribute is added to the ToolsPanel UI
2. And a source value has been set to the binding.

Basically, the help text would appear only after a source has been set. Instead, on trunk it appears immedialy after an attribute is added. As such, I think we need to use the ToolsPanel mechanism, which assumes a `div` element.

Reply:
I would personally argue that the behavior present in `trunk` isn't entirely valid as you need to explicitly select a binding after exposing the attribute name in the UI, so it might be misleading as of today. Anyway, the proposed fix also works.